test/test_hyper.py

```diff
@@ -275,6 +276,18 @@ def test_closed_connections_are_reset(self):
         assert c.recent_stream is None
         assert c.next_stream_id == 1
         assert c.window_manager is not wm
+        with c._conn as conn:
+            assert conn.state_machine.state == ConnectionState.IDLE
+
+        c = HTTP20Connection('www.google.com')
```

Comment:
@plucury Let me rephrase.

The purpose of this change is to enable some specific behaviour, yes? That means that, before this change, there was presumably some resetting that was not happening, and after this change there is some resetting that *is* happening. We should write a test that confirms this behaviour: namely, that test should fail *before* this patch applies and pass afterward.

Is that true of the test written here? If it isn't, we should rewrite it so that it is.